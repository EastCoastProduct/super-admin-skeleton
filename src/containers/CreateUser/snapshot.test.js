import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { CreateUserComponent } from './';

jest.mock('redux-form/immutable', () => ({
  Field: 'Field',
  reduxForm: () => component => component,
}));

describe('CreateUser component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders required data', () => {
    const tree = renderer.create(
      <CreateUserComponent
        dispatch={() => {}}
        form="CreateUser"
        handleSubmit={f => f => f}
        router={{}}
        submitting={true}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders required data with error', () => {
    const tree = renderer.create(
      <CreateUserComponent
        dispatch={() => {}}
        error="User already exists."
        form="CreateUser"
        handleSubmit={f => f => f}
        router={{}}
        submitting={false}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
