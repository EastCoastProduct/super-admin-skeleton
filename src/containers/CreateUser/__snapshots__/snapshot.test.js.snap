exports[`CreateUser component snapshot renders required data 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    Create New User
  </h1>
  <div
    className="box_5m47zw">
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="CreateUser"
        label="Email"
        name="email"
        placeholder="Email"
        type="email"
        validated={true} />
      <div>
        <button
          className="button_w1w7v0-o_O-button_1q6mnjs"
          disabled={true}
          type="submit">
          Create
        </button>
        <a
          className="link_1mjr7jy"
          onClick={[Function]}
          style={Object {}}>
          Cancel
        </a>
      </div>
    </form>
  </div>
</main>
`;

exports[`CreateUser component snapshot renders required data with error 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    Create New User
  </h1>
  <div
    className="box_5m47zw">
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="CreateUser"
        label="Email"
        name="email"
        placeholder="Email"
        type="email"
        validated={true} />
      <div>
        <button
          className="button_w1w7v0-o_O-button_1q6mnjs"
          disabled={false}
          type="submit">
          Create
        </button>
        <a
          className="link_1mjr7jy"
          onClick={[Function]}
          style={Object {}}>
          Cancel
        </a>
      </div>
      <p
        className="error_8cisgn">
        <i
          className="icon_1w16ibx fa fa-exclamation-triangle" />
        User already exists.
      </p>
    </form>
  </div>
</main>
`;
